packages/react-core/src/components/TextInput/TextInput.tsx

```diff
@@ -180,7 +180,7 @@ export class TextInputBase extends React.Component<TextInputProps> {
           styles.formControl,
           validated === ValidatedOptions.success && styles.modifiers.success,
           validated === ValidatedOptions.warning && styles.modifiers.warning,
-          (iconVariant || customIconUrl) && styles.modifiers.icon,
+          (iconVariant !== 'search' || customIconUrl) && styles.modifiers.icon,
```

Comment:
Also needs a check to see if it is defined otherwise it will always add the icon style
```suggestion
          ((iconVariant && iconVariant !== 'search') || customIconUrl) && styles.modifiers.icon,
```